--- mod_opusfile.c	2020-01-05 13:56:50.000000000 -0500
+++ mod_opusfile.c.new	2020-01-05 13:50:35.000000000 -0500
@@ -148,7 +148,7 @@
 			if (globals.debug) {
 				switch_log_printf(SWITCH_CHANNEL_LOG, SWITCH_LOG_DEBUG, "[OGG/OPUS Decoder]: EOF reached [%d]\n", ret);
 			}
-			context->eof = TRUE;
+			context->eof = SWITCH_TRUE;
 			break;
 		} else /* (ret > 0)*/ {
 			/*The number of samples read per channel on success*/
@@ -161,7 +161,7 @@
 		}
 	}
 	switch_mutex_unlock(context->audio_mutex);
-	context->eof = FALSE; // for next page 
+	context->eof = SWITCH_FALSE; // for next page 
 	return SWITCH_STATUS_SUCCESS;
 }
 
@@ -282,7 +282,6 @@
 			switch_log_printf(SWITCH_CHANNEL_LOG, SWITCH_LOG_INFO, "[OGG/OPUS File] Channels: %i\n", head->channel_count);
 			if (head->input_sample_rate) {
 				switch_log_printf(SWITCH_CHANNEL_LOG, SWITCH_LOG_INFO, "[OGG/OPUS File] Original sampling rate: %lu Hz\n", (unsigned long)head->input_sample_rate);
-				handle->samplerate = context->samplerate = head->input_sample_rate;
 			}
 		}
 		if (op_seekable(context->of)) {
@@ -365,14 +364,14 @@
 
 	if (!handle->handler) {
 		if (switch_opusfile_decode(context, data, bytes, handle->real_channels) == SWITCH_STATUS_FALSE) {
-			context->eof = 1;
+			context->eof = SWITCH_TRUE;
 		}
 	}
 	switch_mutex_lock(context->audio_mutex);
 	rb = switch_buffer_read(context->audio_buffer, data, bytes);
 	switch_mutex_unlock(context->audio_mutex);
 
-	if (!rb && (context->eof)) {
+	if (!rb || (context->eof)) {
 		return SWITCH_STATUS_FALSE;
 	}
 	if (rb) {
@@ -530,4 +529,3 @@
  * For VIM:
  * vim:set softtabstop=4 shiftwidth=4 tabstop=4 noet:
  */
-
